design/open/lakefs-on-kv-testing-plan.md

```diff
@@ -0,0 +1,128 @@
+# lakeFS on KV - Testing Plan
+
+## Global Testing
+
+### Migration
+* Global tests to verify and benchmark the migration process
+* Need to verify that the system behavior is not affected by the move to the KV DB
+* [Question] - Do we need to dispose of these once migration to KV is globally completed?
+
+#### Operational Testing
+* Set of tests that will run DB migration as part of the test
+* Currently, system tests assume lakeFS is up and running. Need to add ability to perform migration in the middle of the test (stop lakeFS -> run&verify migration -> start lakeFS ?)
+* Maybe something that can be 
+
+#### Data Level Testing
+* Dump (all) data before and after migration and compare it
+* Verifies data is preserved through migration, and is readable by the KV Store
+* Need to write appropriate dumpers. Can be done per package, as we go
+* Can be done both on DB package level, to verify the package itself, and on the using packages level, to verify data usage is not affected
+* [Question] - How do we define a covering data set, to test? How do we create
+
+### Performance
+* **A benchmark for the migration process itself:**
+  * How do we load the DB with data to migrate for benchmarking?
+  * Ideally, run on several data sizes (tens / thousands / millions branches & commits, various numbers of users, etc.) to get sense of how scaling will affect
+  * What is the expected/acceptable outcome?
```

Comment:
When and how do we run it?

Reply:
A dedicated system-wide migration tests, as part of `Esti` seems like the proper place
WDYT?

Reply:
👍 
Let's add it then

Reply:
Added